src/Sylius/Bundle/CoreBundle/DiscountApplicationCriteria/DiscountApplicationCriteriaInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DiscountApplicationCriteria;
+
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+
+interface DiscountApplicationCriteriaInterface
+{
+    public function isApplicable(CatalogPromotionInterface $catalogPromotion, CatalogPromotionActionInterface $action, ChannelPricingInterface $channelPricing): bool;
```

Comment:
I'm wondering about the last argument. I'm not a very big fan of this, but **maybe** it would be worth using there `array $context = []`? As we see with `ExclusiveCriteria`, not all of the criteria must be ChannelPricing-based. With this approach, we can also move the interface (and `ExclusiveCriteria` implementation) to **PromotionBundle** and make it more generic. As said before, I don't like too much the loose argument approach it uses... but maybe it would be valid in this specific case 🖖

Reply:
I'm also in favor of using the array $context here since indeed, some custom made discount application criteria may rely on many different things

Reply:
indeed, i moved this interface to PromotionBundle and added array $context as a second argument 🖖